Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,760 @@
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <sofa/component/engine/select/BaseROI.h>
+#include <sofa/core/visual/VisualParams.h>
+#include <sofa/type/BoundingBox.h>
+#include <limits>
+#include <sofa/core/topology/BaseTopology.h>
+#include <sofa/core/loader/MeshLoader.h>
+#include <sofa/helper/accessor.h>
+
+namespace sofa::component::engine::select
+{
+
+using core::behavior::BaseMechanicalState ;
+using core::topology::TopologyContainer ;
+using core::topology::BaseMeshTopology ;
+using core::objectmodel::ComponentState ;
+using core::objectmodel::BaseData ;
+using core::objectmodel::Event ;
+using core::loader::MeshLoader ;
+using core::ExecParams ;
+using type::TBoundingBox ;
+using type::Vec3 ;
+using type::Vec4f ;
+using helper::WriteOnlyAccessor ;
+using helper::ReadAccessor ;
+using type::vector ;
+
+template <class DataTypes>
+BaseROI<DataTypes>::BaseROI()
+    : d_positions( initData (&d_positions, "position", "Rest position coordinates of the degrees of freedom. \n"
+                                         "If empty the positions from a MechanicalObject then a MeshLoader are searched in the current context. \n"
+                                         "If none are found the parent's context is searched for MechanicalObject." ) )
+    , d_edges(initData (&d_edges, "edges", "Edge Topology") )
+    , d_triangles(initData (&d_triangles, "triangles", "Triangle Topology") )
+    , d_quads(initData (&d_quads, "quads", "Quad Topology") )
+    , d_tetrahedra(initData (&d_tetrahedra, "tetrahedra", "Tetrahedron Topology") )
+    , d_hexahedra(initData (&d_hexahedra, "hexahedra", "Hexahedron Topology") )
+    , d_computeEdges( initData(&d_computeEdges, true,"computeEdges","If true, will compute edge list and index list inside the ROI. (default = true)") )
```

Comment:
```suggestion
    , d_computeEdges( initData(&d_computeEdges, true,"computeEdges","If true, will compute edge list and index list inside the ROI.") )
```
You can remove all the mentions to the default value as Data now have knowledge of its default value. See https://alxbilger.github.io/doc/components/engine/select/boxroi/ for example